### DISCUSSION TITLE
Integer values in XML configuration files

### DISCUSSION THREAD

**Discussion by jorgerucoba** [symfony/symfony#43037] [Q&A]
Hello. I have a problem with XML configuration files and data types:
I have this constraint
```
<constraint name="Choice">
    <option name="choices">
        <value>1</value>
        <value>2</value>
        <value>3</value>
        <value>4</value>
    </option>
</constraint>
```
With a json payload `"state": 4`  , doesn’t work , but with `"state": "4"` , it does.
Is there any way to explicitly tell the XML that the value is of type integer?
Thanks for your kind help.

**Comment by jfsenechal** [comment]
Hi

Maybe here : https://symfony.com/doc/current/reference/constraints/Choice.html

**Comment by javiereguiluz** [comment]
In this section of the docs, if you click on the XML tab, you'll see the different types of parameters you can define in XML config:

https://symfony.com/doc/current/configuration.html#configuration-parameters

This also matches the related XSD file config:

https://github.com/symfony/symfony/blob/6c2608b1f989164cf2307291b1259019f1866d88/src/Symfony/Component/DependencyInjection/Loader/schema/dic/services/services-1.0.xsd#L297-L304

So, it looks like you can't define integers in XML file ... but I might be wrong.
